### PR TITLE
databases-week2/NicoQ

### PR DESCRIPTION
**Pull request by lnknfan90**
Added the homework, thanks for the review :)

### CODE REVIEW

File: 5_databases/week2/Part1.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-- Active: 1699188605886@@0.0.0.0@3306@hyf_lesson1
+
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (36, 'Catch every Pokemon', 'Catch them all', '1997-04-01 06:54:16', '2023-03-24 13:05:09', null, 2, 3);
+
+UPDATE task
+SET title = "Become a trillionaire"
+WHERE title = "Become a billionaire"
```
Comment (adionditsak) [inline 5_databases/week2/Part1.sql:7]: Correct, but better to do with a primary key (id) so you dont edit more than one

File: 5_databases/week2/Part2.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    PRIMARY KEY (id)  
+) ENGINE=innoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+CREATE TABLE Student (
+    id int(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    name VARCHAR(255) NOT NULL,
+    email VARCHAR(255) NULL,
+    phone VARCHAR(20) NULL,
+    class_id INT,
```
Comment (adionditsak) [inline 5_databases/week2/Part2.sql:16]: Use FOREIGN KEY to indicate what id in class you refer to

File: 5_databases/week2/Part3.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+--Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT *
+FROM task JOIN `user` on email LIKE "%@spotify.com"
```
Comment (adionditsak) [inline 5_databases/week2/Part3.sql:3]: remember to add `;` after all queries

File: 5_databases/week2/Part3.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+--Get all the tasks for 'Donald Duck' with status 'Not started'
+
+SELECT task.title, user.name, status_id
```
Comment (adionditsak) [inline 5_databases/week2/Part3.sql:7]: You want to include user_task table to connect the many_to_many relations here to get the correct response

File: 5_databases/week2/Part3.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+FROM task JOIN `user` on name = "Donald Duck"
+Where status_id = 1
+
+--Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+
+SELECT task.title, user.name, created
```
Comment (adionditsak) [inline 5_databases/week2/Part3.sql:13]: You want to include user_task table to connect the many_to_many relations here to get the correct response

File: 5_databases/week2/Part3.sql [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+FROM task JOIN `user` on name = "Maryrose Meadows"
+WHERE month(created)= 09
+
+--Find how many tasks where created in each month, e.g. how many tasks were created in october, how many tasks were created in november, etc. (hint: use group by)
+SELECT MONTH (created), COUNT(*)
+FROM task
+GROUP BY MONTH(created)
```
Comment (adionditsak) [inline 5_databases/week2/Part3.sql:20]: Remove space between MONTH and (created).

Good job!

File: 5_databases/week2/Part3.sql (hunk 3 above)
Comment (adionditsak) [reply 5_databases/week2/Part3.sql:3]: Here you should use the user_task many-to-many table to make relations so you get the task and not the user

**Comment by adionditsak** [comment]
@lnknfan90 How is it going, let me know if you need help?

File: 5_databases/week2/Part3.sql (hunk 5 above)
Comment (lnknfan90) [reply 5_databases/week2/Part3.sql:13]: oops, I forgot to add the user_task table, thanks for pointing it out to me :)

**Comment by lnknfan90** [comment]
> @lnknfan90 How is it going, let me know if you need help?

- All good! Sorry for the late check, I couldn't check this assignment before.
Thanks for your review, was important, because I miss one table, so was a very different query to do, and I didn't notice before you mentioned. Have a good weekend!

**Pull request closed without merging**